File: src/types/file.ts
export interface FileEntry {
  name: string;
  path: string;
  is_dir: boolean;
  extension: string | null;
  size: number | null;
  modified: string | null;
  children: FileEntry[] | null;
}

export interface FileContent {
  path: string;
  name: string;
  content: string;
  modified: string | null;
  size: number;
}

File: src-tauri/src/models/file.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FileEntry {
    pub name: String,
    pub path: String,
    pub is_dir: bool,
    pub extension: Option<String>,
    pub size: Option<u64>,
    pub modified: Option<String>,
    pub children: Option<Vec<FileEntry>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FileContent {
    pub path: String,
    pub name: String,
    pub content: String,
    pub modified: Option<String>,
    pub size: u64,
}
